pyrk/input_uncert.py

```diff
@@ -0,0 +1,174 @@
+''' changed from the input file input_FVM.py
+to conduct uncertainty study
+'''
+from ur import units
+import th_component as th
+import math
+from cool_uncert import Cool
+from mod import Mod
+from fuel import Fuel
+from timer import Timer
+import random
+#############################################
+#
+# User Workspace
+#
+#############################################
+
+# Thermal hydraulic params
+# Temperature feedbacks of reactivity
+alpha_fuel = random.gauss(-3.19, 0.05*3.8)*units.pcm/units.kelvin
+alpha_cool = random.gauss(0.23, 0.11)*units.pcm/units.kelvin
+alpha_mod = -0.7*units.pcm/units.kelvin
+alpha_shell = 0*units.pcm/units.kelvin
+
+#initial values for temepratures
+t_mod = (800+273.15)*units.kelvin
+t_fuel = (800+273.15)*units.kelvin
+t_shell = (770+273.15)*units.kelvin
+t_cool = (650+273.15)*units.kelvin
+
+kappa = 0.0
+
+# Initial time
+t0 = 0.00*units.seconds
+
+# Timestep
+dt = 0.1*units.seconds
+
+# Final Time
+tf = 80*units.seconds
+
+
+def area_sphere(r):
+    assert(r >= 0*units.meter)
+    return (4.0)*math.pi*pow(r.to('meter'), 2)
+
+
+def vol_sphere(r):
+    assert(r >= 0*units.meter)
+    return (4./3.)*math.pi*pow(r.to('meter'), 3)
+
+# volumes
+n_pebbles = 470000
+r_mod = 1.25*units.centimeter
+r_fuel = 1.4*units.centimeter
+r_shell = 1.5*units.centimeter
+
+vol_mod = vol_sphere(r_mod)
+vol_fuel = vol_sphere(r_fuel) - vol_sphere(r_mod)
+vol_shell = vol_sphere(r_shell) - vol_sphere(r_fuel)
+vol_cool = (vol_mod + vol_fuel + vol_shell)*0.4/0.6
+a_pb = area_sphere(r_shell)
+
+# 4700TODO implement h(T) model
+h_cool = 4700.0*units.watt/units.kelvin/units.meter**2
+m_flow = 976*units.kg/units.second  # 976*units.kg/units.second
+t_inlet = units.Quantity(600.0, units.degC)  # degrees C
+
+#############################################
+#
+# Required Input
+#
+#############################################
+
+# Total power, Watts, thermal
+power_tot = 234000000.0*units.watt
+#power_tot = 0.0*units.watt
+
+# Timer instance, based on t0, tf, dt
+ti = Timer(t0=t0, tf=tf, dt=dt)
+
+# Number of precursor groups
+n_pg = 6
+
+# Number of decay heat groups
+n_dg = 0
+
+# Fissioning Isotope
+fission_iso = "u235"
+
+# Spectrum
+spectrum = "thermal"
+
+# Feedbacks, False to turn reactivity feedback off. True otherwise.
+feedback = True
+
+# External Reactivity
+#from reactivity_insertion import ImpulseReactivityInsertion
```

Comment:
Let me clean these files now

On Tue, Aug 25, 2015 at 11:19 AM, Katy Huff <notifications@github.com>
wrote:

> In pyrk/input_uncert.py
> <https://github.com/pyrk/pyrk/pull/10#discussion_r37900010>:
>
> > +n_pg = 6
> > +
> > +# Number of decay heat groups
> > +n_dg = 0
> > +
> > +# Fissioning Isotope
> > +fission_iso = "u235"
> > +
> > +# Spectrum
> > +spectrum = "thermal"
> > +
> > +# Feedbacks, False to turn reactivity feedback off. True otherwise.
> > +feedback = True
> > +
> > +# External Reactivity
> > +#from reactivity_insertion import ImpulseReactivityInsertion
>
> commented-out code (distinct from comments being used appropriately, for
> information to the user) must be deleted.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyrk/pyrk/pull/10/files#r37900010>.
>



-- 
-----------------------------------------
Xin WANG | 王昕
PhD Student
Thermal Hydraulics Laboratory
Nuclear Engineering Department
University of California, Berkeley
(415)408-1691